android/src/main/java/com/google/android/cameraview/Camera1.java

```diff
@@ -693,14 +739,25 @@ public Size getPreviewSize() {
      * This rewrites {@link #mCameraId} and {@link #mCameraInfo}.
      */
     private void chooseCamera() {
-        for (int i = 0, count = Camera.getNumberOfCameras(); i < count; i++) {
-            Camera.getCameraInfo(i, mCameraInfo);
-            if (mCameraInfo.facing == mFacing) {
-                mCameraId = i;
-                return;
+        if(_mCameraId == null){
+            for (int i = 0, count = Camera.getNumberOfCameras(); i < count; i++) {
+                Camera.getCameraInfo(i, mCameraInfo);
+                if (mCameraInfo.facing == mFacing) {
+                    mCameraId = i;
+                    return;
+                }
+            }
+            mCameraId = 0; // default to the only available camera
```

Comment:
@sibelius  Can you comment on these lines? I wonder if we need to do something different here actually to handle the case where a camera is not found.  I think I overlooked this one and it should have been INVALID_CAMERA_ID instead. Thoughts?
Need to test on a device with no cameras, or just 1 camera.

Reply:
hard to find a device without cameras

what happen when you throw? can an user get this error using an ErrorBoundary?

`0` is always the first camera?

won't break the current behavior?

Reply:
We can/should probably mimic what Camera2 does and raise an error if there are 0 cameras, but set the camera to the first one available if we didn't find one: https://github.com/react-native-community/react-native-camera/blob/master/android/src/main/java/com/google/android/cameraview/Camera2.java#L688

How does that sound? I believe the Camera1 code originally would just set the camera ID to be the invalid one and you would get no image at all in this case (if lets say you don't have a back camera). It's a very specific case.

I was thinking about something like this:

```
int count = Camera.getNumberOfCameras();
            if(count == 0){
                throw new RuntimeException("No camera available.");
            }

            for (int i = 0; i < count; i++) {
                Camera.getCameraInfo(i, mCameraInfo);
                if (mCameraInfo.facing == mFacing) {
                    mCameraId = i;
                    return;
                }
            }
            // no camera found, set the one we have
            mCameraId = 0;
            Camera.getCameraInfo(mCameraId, mCameraInfo);
```

Reply:
this part is weird

// no camera found, set the one we have
mCameraId = 0;

but it makes sense, you are looking for the camera facing the correct facing the user selected

Reply:
The original code would simply set the camera to be the INVALID_ID one, but that behaviour is different from the Camera2 API which simply sets the first available camera if the code wasn't able to find the back/front facing camera. Do you think it makes sense to make both APIs consistent?

Reply:
sure

Reply:
I've just added a commit for that.